lib/modules.nix

```diff
@@ -87,6 +84,16 @@ rec {
       result = { inherit options config; };
     in result;
 
+
+ /* Filter disabled modules. Modules can be disabled allowing
+    their implementation to be replaced. */
+ filterModules = modules:
+   let
+     moduleKey = m: if isString m then toString ../nixos/modules + "/" + m else toString m;
```

Comment:
If you can find a way such that one can move this default path as an argument of the module system instead of as a static string in it, this might be preferable.

An example, even without leaving NixOS, is for sub-modules which are defined in different paths, should we name the complete NixOS module path within NixOS, or just name it as a relative file, as this trick is doing for NixOS modules.